lib/dialects/abstract/query-generator.js

```diff
@@ -1386,16 +1386,34 @@ const QueryGenerator = {
         }
         mainQueryItems.push(this.selectFromTableFragment(options, model, mainAttributes, '('+
           options.groupedLimit.values.map(value => {
-            const where = _.assign({}, options.where);
-            where[options.groupedLimit.on] = value;
+            let where = options.where || {}
+              , include;
+
+            if (Array.isArray(options.groupedLimit.on)) {
+              include = Model._validateIncludedElements({
+                include: [{
+                  attributes: [],
+                  association: options.groupedLimit.on[0],
+                  where: {
+                    [options.groupedLimit.on[1]]: value
+                  }
+                }],
+                model
+              }).include;
+            } else {
+              where = _.assign(options.where);
+              where[options.groupedLimit.on] = value;
+            }
```

Comment:
We have tests for grouped limit with top level include though https://github.com/sequelize/sequelize/blob/080b3f8511027107a6dbaeeb0f3ee4a22451c101/test/unit/sql/select.test.js#L174 (Not saying its ever used in pratice, but we support it :P)